File: frontend/src/types/adsAlert.ts
// Types for the Ads Alert promotional messaging system

export type PromotionStatus = 'draft' | 'scheduled' | 'sent' | 'cancelled'
export type MediaType = 'text' | 'image' | 'video' | 'document' | 'mixed'
export type TargetType = 'all' | 'selected'
export type BroadcastStatus = 'pending' | 'sent' | 'failed'

// Chat types
export interface Chat {
  id: string
  tenant_id: string
  platform: string
  chat_id: string
  chat_name?: string
  customer_name?: string
  tags: string[]
  subscribed: boolean
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface ChatCreate {
  platform?: string
  chat_id: string
  chat_name?: string
  customer_name?: string
  tags?: string[]
}

export interface ChatUpdate {
  chat_name?: string
  customer_name?: string
  tags?: string[]
  subscribed?: boolean
}

// Promotion types
export interface Promotion {
  id: string
  tenant_id: string
  title: string
  content?: string
  status: PromotionStatus
  media_urls: string[]
  media_type: MediaType
  scheduled_at?: string
  target_type: TargetType
  target_chat_ids: string[]
  sent_at?: string
  created_by?: string
  created_at: string
  updated_at: string
}

export interface PromotionCreate {
  title: string
  content?: string
  media_urls?: string[]
  media_type?: MediaType
  target_type?: TargetType
  target_chat_ids?: string[]
  scheduled_at?: string
}

export interface PromotionUpdate {
  title?: string
  content?: string
  media_urls?: string[]
  media_type?: MediaType
  target_type?: TargetType
  target_chat_ids?: string[]
  scheduled_at?: string
  status?: PromotionStatus
}

export interface PromotionScheduleRequest {
  scheduled_at: string
}

// Folder types
export interface MediaFolder {
  id: string
  tenant_id: string
  name: string
  parent_id?: string
  created_by?: string
  created_at: string
}

export interface FolderCreate {
  name: string
  parent_id?: string
}

export interface FolderTreeNode {
  id: string
  name: string
  parent_id?: string
  children: FolderTreeNode[]
}

// Media types
export interface MediaFile {
  id: string
  tenant_id: string
  folder_id?: string
  filename: string
  original_filename?: string
  storage_path: string
  url: string
  file_type: string
  file_size?: number
  thumbnail_url?: string
  width?: number
  height?: number
  duration?: number
  created_by?: string
  created_at: string
}

export interface MediaUploadResponse {
  id: string
  filename: string
  storage_path: string
  url: string
  file_type: string
  file_size: number
}

export interface MediaDeleteResponse {
  id: string
  filename: string
  deleted: boolean
}

// Broadcast types
export interface BroadcastResult {
  chat_id: string
  success: boolean
  error?: string
}

export interface BroadcastResponse {
  promotion_id: string
  total: number
  sent: number
  failed: number
  results: BroadcastResult[]
}

// Stats types
export interface AdsAlertStats {
  total_chats: number
  subscribed_chats: number
  total_promotions: number
  draft_promotions: number
  scheduled_promotions: number
  sent_promotions: number
  total_media: number
  total_folders: number
}

// List params
export interface ChatListParams {
  subscribed_only?: boolean
  tags?: string
  limit?: number
}

export interface PromotionListParams {
  status?: PromotionStatus
  limit?: number
}

export interface MediaListParams {
  folder_id?: string
  file_type?: string
  limit?: number
}

File: app/schemas/ads_alert.py
# app/schemas/ads_alert.py
"""Pydantic schemas for Ads Alert system"""
from typing import List, Optional
from datetime import datetime
from uuid import UUID
from pydantic import BaseModel, Field
from enum import Enum


# Enums for API
class PromotionStatusEnum(str, Enum):
    draft = "draft"
    scheduled = "scheduled"
    sent = "sent"
    cancelled = "cancelled"


class MediaTypeEnum(str, Enum):
    text = "text"
    image = "image"
    video = "video"
    document = "document"
    mixed = "mixed"


class TargetTypeEnum(str, Enum):
    all = "all"
    selected = "selected"


class BroadcastStatusEnum(str, Enum):
    pending = "pending"
    sent = "sent"
    failed = "failed"


# Chat Schemas
class ChatCreate(BaseModel):
    """Create a new chat registration"""
    platform: str = "telegram"
    chat_id: str = Field(..., min_length=1, max_length=100)
    chat_name: Optional[str] = None
    customer_name: Optional[str] = None
    tags: List[str] = []


class ChatUpdate(BaseModel):
    """Update chat registration"""
    chat_name: Optional[str] = None
    customer_name: Optional[str] = None
    tags: Optional[List[str]] = None
    subscribed: Optional[bool] = None


class ChatResponse(BaseModel):
    """Chat response model"""
    id: UUID
    tenant_id: UUID
    platform: str
    chat_id: str
    chat_name: Optional[str]
    customer_name: Optional[str]
    tags: List[str]
    subscribed: bool
    is_active: bool
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# Promotion Schemas
class PromotionCreate(BaseModel):
    """Create a new promotion"""
    title: str = Field(..., min_length=1, max_length=255)
    content: Optional[str] = None
    media_urls: List[str] = []
    media_type: MediaTypeEnum = MediaTypeEnum.text
    target_type: TargetTypeEnum = TargetTypeEnum.all
    target_chat_ids: List[UUID] = []
    scheduled_at: Optional[datetime] = None


class PromotionUpdate(BaseModel):
    """Update promotion"""
    title: Optional[str] = Field(None, min_length=1, max_length=255)
    content: Optional[str] = None
    media_urls: Optional[List[str]] = None
    media_type: Optional[MediaTypeEnum] = None
    target_type: Optional[TargetTypeEnum] = None
    target_chat_ids: Optional[List[UUID]] = None
    scheduled_at: Optional[datetime] = None
    status: Optional[PromotionStatusEnum] = None


class PromotionResponse(BaseModel):
    """Promotion response model"""
    id: UUID
    tenant_id: UUID
    title: str
    content: Optional[str]
    status: PromotionStatusEnum
    media_urls: List[str]
    media_type: MediaTypeEnum
    scheduled_at: Optional[datetime]
    target_type: TargetTypeEnum
    target_chat_ids: List[UUID]
    sent_at: Optional[datetime]
    created_by: Optional[UUID]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class PromotionSendRequest(BaseModel):
    """Request to send a promotion immediately"""
    pass  # No additional fields needed


class PromotionScheduleRequest(BaseModel):
    """Request to schedule a promotion"""
    scheduled_at: datetime


# Media Folder Schemas
class FolderCreate(BaseModel):
    """Create a new folder"""
    name: str = Field(..., min_length=1, max_length=255)
    parent_id: Optional[UUID] = None


class FolderResponse(BaseModel):
    """Folder response model"""
    id: UUID
    tenant_id: UUID
    name: str
    parent_id: Optional[UUID]
    created_by: Optional[UUID]
    created_at: datetime

    class Config:
        from_attributes = True


class FolderTreeResponse(BaseModel):
    """Folder with children for tree view"""
    id: UUID
    name: str
    parent_id: Optional[UUID]
    children: List["FolderTreeResponse"] = []

    class Config:
        from_attributes = True


# Update forward reference
FolderTreeResponse.model_rebuild()


# Media Schemas
class MediaResponse(BaseModel):
    """Media file response model"""
    id: UUID
    tenant_id: UUID
    folder_id: Optional[UUID]
    filename: str
    original_filename: Optional[str]
    storage_path: str
    url: str  # Computed public URL
    file_type: str
    file_size: Optional[int]
    thumbnail_url: Optional[str]  # Computed thumbnail URL
    width: Optional[int]
    height: Optional[int]
    duration: Optional[int]
    created_by: Optional[UUID]
    created_at: datetime

    class Config:
        from_attributes = True


class MediaUploadResponse(BaseModel):
    """Response after uploading media"""
    id: UUID
    filename: str
    storage_path: str
    url: str
    file_type: str
    file_size: int


class MediaDeleteResponse(BaseModel):
    """Response after deleting media"""
    id: UUID
    filename: str
    deleted: bool


# Broadcast Schemas
class BroadcastRequest(BaseModel):
    """Request to broadcast to multiple chats"""
    chat_ids: List[str]  # Telegram chat IDs (strings)
    content: str
    media_type: MediaTypeEnum = MediaTypeEnum.text
    media_urls: List[str] = []


class BroadcastResult(BaseModel):
    """Result of broadcasting to a single chat"""
    chat_id: str
    success: bool
    error: Optional[str] = None


class BroadcastResponse(BaseModel):
    """Response after broadcasting"""
    promotion_id: UUID
    total: int
    sent: int
    failed: int
    results: List[BroadcastResult]


class BroadcastLogResponse(BaseModel):
    """Broadcast log entry response"""
    id: UUID
    promotion_id: UUID
    chat_id: UUID
    status: BroadcastStatusEnum
    error_message: Optional[str]
    sent_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True


# Stats Schemas
class AdsAlertStats(BaseModel):
    """Statistics for ads alert system"""
    total_chats: int
    subscribed_chats: int
    total_promotions: int
    draft_promotions: int
    scheduled_promotions: int
    sent_promotions: int
    total_media: int
    total_folders: int
